x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinatorTests.java

```diff
@@ -2350,6 +1929,150 @@ public void testFollowerIndexIsCreatedInExecuteAutoFollow() {
         assertThat(autoFollowResults.v2().contains(indexName), equalTo(true));
     }
 
+    public void testRemovesClusterLevelErrorsOnRemovingAutoFollowPattern() {
+        // given auto-follow pattern added
+        var pattern1 = createAutoFollowPattern("remote1", "logs-*");
+        var pattern2 = createAutoFollowPattern("remote2", "logs-*");
+        var pattern3 = createAutoFollowPattern("remote2", "metrics-*");// same remote
+
+        var autoFollowCoordinator = createAutoFollowCoordinator();
+        autoFollowCoordinator.updateAutoFollowers(
+            createClusterStateWith(Map.of("pattern1", pattern1, "pattern2", pattern2, "pattern3", pattern3))
+        );
+
+        // and stats are published
+        autoFollowCoordinator.updateStats(
+            List.of(
+                new AutoFollowCoordinator.AutoFollowResult("pattern1", new RuntimeException("ClusterStateFetchException")),
+                new AutoFollowCoordinator.AutoFollowResult("pattern2", new RuntimeException("ClusterStateFetchException")),
+                new AutoFollowCoordinator.AutoFollowResult("pattern3", new RuntimeException("ClusterStateFetchException"))
+            )
+        );
+
+        // when auto-follow pattern `pattern3` is removed
+        var before = autoFollowCoordinator.getStats();
+        autoFollowCoordinator.updateAutoFollowers(createClusterStateWith(Map.of("pattern1", pattern1, "pattern2", pattern2)));
+        autoFollowCoordinator.updateStats(List.of());// actually triggers the purge
+        var after = autoFollowCoordinator.getStats();
+
+        // then stats are removed as well (but only for the removed pattern)
+        assertThat(before.getRecentAutoFollowErrors().keySet(), contains("pattern1", "pattern2", "pattern3"));
+        assertThat(after.getRecentAutoFollowErrors().keySet(), allOf(contains("pattern1", "pattern2"), not(contains("pattern3"))));
```

Comment:
This is slightly confusing. `contains` verifies same length and same order, which gives rise to two comments:

1. I think the order is not relevant. So `equalTo` a `Set.of` would be better here, alternatively `containsInAnyOrder`. The test only passes (I think) because of the use of `LinkedHashMap` internally.
2. The `not(contains("pattern3"))` is true  because it is not a list containing only that one element.

I think we should simply use `equalTo` instead.

I think this comment is true for the other `contains` verifications below in this PR.